integration-tests/gradle/src/main/resources/dependency-contraints-project/build.gradle

```diff
@@ -0,0 +1,30 @@
+plugins {
+    id 'java'
+    id 'io.quarkus'
+}
+
+group = 'com.quarkus.demo'
+version = '1.0'
+
+repositories {
+    mavenCentral()
+    if (System.properties.containsKey('maven.repo.local')) {
+        maven {
+            url System.properties.get('maven.repo.local')
+        }
+    } else {
+        mavenLocal()
+    }
+}
+
+dependencies {
+    implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
+    implementation 'io.quarkus:quarkus-resteasy'
+
+    constraints {
+        implementation 'javax.json.bind:javax.json.bind-api:1.0'
```

Comment:
How do we know this constraint is effective?

Reply:
The build was failing because it was not able to resolve the dependency without the version constraint. 
I can add a check to validate the version in the output folder to make sure everything is fine

Reply:
That would be good. Thanks @glefloch

Reply:
Why would it fail though w/o the constraint?

Reply:
Ah, ok, sorry, i see it.

Reply:
@aloubyansky I added a check to validate the jar is present in the correct version in the output directory